### ISSUE TITLE
OA URL cannot be updated on Admin

### ISSUE THREAD

**Issue by DanCoughlin** [psu-stewardship/researcher-metadata#190]
The admin UI does not have the ability to update the open access url that is obtained from Open Access Button. In some cases these values are wrong and it would be helpful if the admins could update (or delete) the URL.

**Comment by ajkiessl** [comment]
We should skip over OAB imports when the open_access_button_last_checked_at timestamp has been set and open_access_url is blank.  This indicates that an admin deleted the open_access_url.

Since the OAB import already ignores Publications that have an open_access_url set, the import will skip any Publications that have simply been edited by an admin (not deleted).

We may need to update the existing logic to not update open_access_button_last_checked_at if the url we receive from OAB is blank.  If we are receiving blank urls _and_ updating the open_access_button_last_checked_at timestamp, these could be skipped over and we would not receive an OA url if one was added in the future.

**Comment by anaelizabethenriquez** [comment]
As I see it, the main goal with this issue is to be able to delete the URL that came from OAB. It's rare that we find errors, but when we find an error, it would be nice to be able to delete it.

In the issue description, Dan mentioned editing this OAB URL value, but I think the only editing we want to do is to remove it. (If I had an OA URL to add that wasn't coming from OAB, I would put it in the User-submitted OA URL field.)

A bit different from what Alex wrote above, I think we should check OAB again for items where an admin has deleted the OAB URL. OAB updates their database frequently -- generally I'd expect them to have fixed the error by the time RMD does the next OAB import. 

Down the road (connected to #199 and #192 ), I anticipate that we'll want RMD to check again on OA URLs. They can change or disappear, for a variety of reasons. Enabling admins to delete them is essentially a way to force RMD to check again, when we know a URL is bad.

**Comment by nicktiberi** [comment]
@anaelizabethenriquez - the current behavior of the OAB import is that we only attempt to update publications that are missing URLs. So if we simply add the ability for the admin to edit the Open Access URL, the following would happen:

- if the URL is deleted entirely, the next time OAB import runs, it will replace the URL
- if the URL is modified in some other way, the next time OAB import runs, it will not update the URL

Based on what you said, since OAB frequently fixes the broken URLs, it doesn't seem like we would need to change the import logic. Do you agree?

**Comment by EricDurante** [comment]
Yeah, I was about to comment along the same lines as Ana, so I totally agree. I'm thinking that the whole business of importing/updating open access URLs/statuses will need to become a lot more dynamic because that information changes a lot more frequently than I initially realized when the OAB import was first implemented.

**Comment by anaelizabethenriquez** [comment]
@nicktiberi Yes, I agree that this issue does not require a change to the current import logic.

In case it's helpful, here's the [Basecamp thread](https://3.basecamp.com/4206247/buckets/15288919/messages/4063918879#__recording_4069643268) where we initially discussed this issue.

**Comment by EricDurante** [comment]
If we want OAB to be able to fix its own errors (we do, I think), then we'll need our importer to update non-blank URLs too. Is that right?

The same concern arises when we have publications in OAB where the open access status has changed. If OAB is telling us that the publication is no longer open access (formerly present OA URL is now gone), then I think we want to remove the URL that we had previously imported.

**Comment by anaelizabethenriquez** [comment]
Yes, Eric, I agree. We should be re-checking non-blank URLs periodically (maybe on every import or maybe less frequently). I don't think that has to be part of this issue, though.

**Comment by ajkiessl** [comment]
If we import the way Eric described, would that eliminate the need to edit the OA URL through the admin interface?

We could modify this ticket a bit and just have Nick tweak the importer so it is more dynamic.

**Comment by EricDurante** [comment]
@ajkiessl That's kind of what I was wondering also. I'd be wary about doing both - that is, giving admins the ability to edit/delete the OAB open access URL _and_ automatically updating it with the importer. That could lead to the situation where an admin makes an edit and then their edit is overwritten, which could be unexpected and possibly undesirable. We handle that situation elsewhere by prohibiting importers from modifying records that have been edited by an admin, but that would be the wrong behavior in the case where an admin deletes a bad URL so that a new (correct) one will be imported.

**Comment by anaelizabethenriquez** [comment]
Yes, I agree that these changes to the importer would make editing the URL less important (and perhaps even undesirable, per Eric's comment above).

I'm not sure of the relation between importer changes and some of the other issues, like  #199 and #192.

**Comment by EricDurante** [comment]
Yeah, they'll all be interrelated in ways that we haven't even fully determined yet. I think the whole OAB importer will need to change considerably if it runs alongside an Unpaywall import.
